app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,59 @@
+class TripsController < ApplicationController
+
+  def index
+    if params[:passenger_id]
+      @passenger = Passenger.find(params[:passenger_id])
+      @trips = @passenger.trips
+    else
+      @trips = Trip.all.order(:date).page params[:page]
+    end
+  end
+
+  def new
+    @trip = Trip.new(passenger_id: params[:passenger_id])
+    @trip.driver_id = Driver.get_random_driver
```

Comment:
Yes - nice job having this method's logic in the model not here in the controller